File: 941-valid-mountain-array/941-valid-mountain-array.c
bool validMountainArray(int* arr, int arrSize){
    int i = 0;

    while(i < arrSize-1 && arr[i] < arr[i + 1])
        i++;

    if(i == 0 || i == arrSize-1)
        return false;

    while(i < arrSize-1 && arr[i] > arr[i + 1])
        i++;

    return (i == arrSize-1);
}

File: 941-valid-mountain-array/941-valid-mountain-array.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        int i = 0;
        
        while(i < arr.length-1 && arr[i+1] > arr[i])
            i++;
        
        if(i == 0 || i == arr.length-1)
            return false;
        
        while(i < arr.length-1 && arr[i] > arr[i+1])
            i++;
        
        return (i == arr.length-1);
    }
}